Fenrir/UbcCheck.fs

```diff
@@ -0,0 +1,283 @@
+module Fenrir.UbcCheck
+
+//https://github.com/cr-marcstevens/sha1collisiondetection/blob/master/lib/ubc_check.c
```

Comment:
Let's please copy-paste the MIT license header from that file right here.

Reply:
should i add it to sha1 file too, there are some code from their repo?

Reply:
Yep. If we had got our code from somewhere else, we have to follow the code's license restrictions. It's a good thing it's all under MIT.